File: Cpp/src/Product.cpp


#include "Price.h"
#include "Product.h"


void Product::save_product_to_database()
{
    printf("Unsupported Operation %s\n",
           "missing from this exercise - shouldn't be called from a unit test");
    exit(1);
}

std::string Product::get_product_name() {
    return _name;
}

std::string Product::get_product_id() {
    return _id;
}

int Product::get_product_weight() {
    return _weight;
}

Price *Product::get_product_price() {
    return _price;
}

bool Product::is_product_event() {
    return false;
}

File: TypeScript/src/Product.ts
import { Price } from "./Price";

export class Product {

    protected readonly name: string;
    protected readonly id: string;
    protected readonly weight: number;
    protected readonly price: Price;

    constructor(name: string, id: string, weight: number, price: Price) {
        this.name = name;
        this.id = id;
        this.weight = weight;
        this.price = price;
    }

    saveToDatabase(): void {
        throw new Error("missing from this exercise - shouldn't be called from a unit test");
    }

    getName(): string {
        return this.name;
    }

    getId(): string {
        return this.id;
    }

    toString(): string {
        return "Product{" + this.name + "}";
    }

    getWeight(): number {
        return this.weight;
    }

    getPrice(): Price {
        return this.price;
    }

    isEvent(): boolean {
        return false;
    }

}
